pkg/proxy/endpoints.go

```diff
@@ -397,41 +402,45 @@ func (em EndpointsMap) LocalReadyEndpoints() map[types.NamespacedName]int {
 	return eps
 }
 
-// detectStaleConnections modifies <staleEndpoints> and <staleServices> with detected stale connections. <staleServiceNames>
-// is used to store stale udp service in order to clear udp conntrack later.
-func detectStaleConnections(oldEndpointsMap, newEndpointsMap EndpointsMap, staleEndpoints *[]ServiceEndpoint, staleServiceNames *[]ServicePortName) {
-	// Detect stale endpoints: an endpoint can have stale conntrack entries if it was receiving traffic
-	// and then goes unready or changes its IP address.
+// detectStaleConntrackEntries detects services that may be associated with stale conntrack entries.
+// (See UpdateEndpointMapResult.DeletedUDPEndpoints and .NewlyActiveUDPServices.)
+func detectStaleConntrackEntries(oldEndpointsMap, newEndpointsMap EndpointsMap, deletedUDPEndpoints *[]ServiceEndpoint, newlyActiveUDPServices *[]ServicePortName) {
```

Comment:
This is fine as a continuation of what we've always done, but the REAL fix would be to load conntrack info and scan that for tuples that don't match and need to be discarded.  It works as is, but if anything goes wrong, we lose the "event" and never come back to reconcile.

Reply:
> This is fine as a continuation of what we've always done, but the REAL fix would be to load conntrack info and scan that for tuples that don't match and need to be discarded.

do you mean to do a diff between the conntrack table and the "virtual generated by kube-proxy expected conntrack table" ?

Reply:
Sort of.  Like "run conntrack -L and look for UDP records which correspond to a service endpoint, but that endpoint doesn't exist in that service" as opposed to a removal "event" .

Reply:
it will be interesting if someone can do the exploratory work to see how expensive is to list all the conntrack table ... some time ago I asked one kernel/netfilter developer to implement server side filtering so you can dump only some specific entries, he told me this will solve the problem http://patchwork.ozlabs.org/project/netfilter-devel/patch/20210730131422.16958-3-fw@strlen.de/ but I never had time to follow up